src/app/static/src/tests/integration/adr-dataset.itest.ts

```diff
@@ -204,4 +205,46 @@ describe("ADR dataset-related actions", () => {
             .toBe("anc.csv");
     }, 7000);
 
+    it("hits upload files to adr endpoint and gets appropriate error", async () => {
+
+        const commit = jest.fn();
+
+        // 1. get datasets
+        await adrActions.getDatasets({commit, rootState} as any);
+
+        // 2. select a naomi dev dataset
+        const datasets = commit.mock.calls[2][0]["payload"];
+        const dataset = datasets.find((dataset: any) => dataset.organization.name === "naomi-development-team");
+
+        // 3. check can upload
+        commit.mockClear();
+        const root = {
+            ...rootState,
+            baseline: {
+                selectedDataset: {
+                    id: dataset.id,
+                    organization: {id: dataset.organization.id}
+                }
+            }
+        };
+
+        const adr = { schemas, datasets };
+
+        const uploadFilesPayload = [
+            {
+                resourceType: "type1",
+                resourceFilename: "file1",
+                resourceId: "id1"
+            }
+        ] as UploadFile[]
+
+        await adrActions.uploadFilestoADR({commit, state: adr, rootState: root} as any,
+            uploadFilesPayload);
+
+        expect(commit.mock.calls.length).toBe(2);
+        expect(commit.mock.calls[0][0]["type"]).toBe("ADRUploadStarted");
+        expect(commit.mock.calls[1][0]["type"]).toBe("SetADRUploadError");
+        expect(commit.mock.calls[1][0]["payload"]["error"]).toBe("OTHER_ERROR");
+    }, 7000);
```

Comment:
This always times out. I'm not sure if there is any way around this besides giving it a really long timer, which is not ideal as it will slow down these tests every time we test a build.

Reply:
I'm not sure if it will succeed or not. Perhaps there is some way to guarantee a fail and get a response quickly from adr. I recall giving it an incorrect `resourceType` produced a quick error response, but I would imagine `"type1"` is an incorrect `resourceType`.

Reply:
I think this should always fail won't it? It's got nonsense test data that won't correspond to any real files. I'd expect it to fail immediately in the backend because the resourceType isn't recognised: https://github.com/mrc-ide/hint/blob/master/src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ADRController.kt#L184
So it shouldn't ever get as far as ADR and shouldn't require a long timeout.

Reply:
Also, if you are just  going for the failure case, you don't need all that set-up with getting a real dataset, you can just give the adr state a dummy dataset too.

Reply:
Oh, in fact, that's probably why it's timing out, it should fly if you get rid of all that.